src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -5,217 +5,16 @@ module FSharp.Compiler.BuildGraph
 open System
 open System.Threading
 open System.Threading.Tasks
-open System.Diagnostics
 open System.Globalization
-open FSharp.Compiler.DiagnosticsLogger
-open Internal.Utilities.Library
 
-[<NoEquality; NoComparison>]
-type NodeCode<'T> = Node of Async<'T>
+type Async with
 
-let wrapThreadStaticInfo computation =
-    async {
-        let diagnosticsLogger = DiagnosticsThreadStatics.DiagnosticsLogger
```

Comment:
NodeCode does this stacking and unwinding implicitly. Without it, we need to be explicit about it by using functions like `CompilationGlobalsScope`, `UseDiagnistcsLogger` etc. Most compiler tasks do use them anyway, so there is little need to alter the existing code.

Reply:
I thought this was the reason why the `AsyncLocal<..>` replacement was used, is it not?

Reply:
Yes, what I meant was: on the boundary, for example `Async.AwaitNodeCode` does this wrapping implicitly.

OTOH AsyncLocal context will always pass across continuations.

Reply:
To clarify further, there are two orthogonal things here, as I understand it:
* Passing loggers between threads in the flow of an async computation. It's what `NodeCode` / `AsyncLocal` does.

 * Stacking / unwinding of diagnostics context. This is not really related to async, single threaded compiler tasks can do it. This is generally done explicitly with `UseDiagnosticsLogger` and such, but incidentally `NodeCode` does this in `AwaitNodeCode`, `AwaitAsync`, so it's possible we rely on this behavior somewhere.